internal/apis/certmanager/validation/certificate.go

```diff
@@ -54,11 +54,25 @@ func ValidateCertificateSpec(crt *internalcmapi.CertificateSpec, fldPath *field.
 
 	var commonName = crt.CommonName
 	if crt.LiteralSubject != "" {
-
 		if !utilfeature.DefaultFeatureGate.Enabled(feature.LiteralCertificateSubject) {
 			el = append(el, field.Forbidden(fldPath.Child("literalSubject"), "Feature gate LiteralCertificateSubject must be enabled on both webhook and controller to use the alpha `literalSubject` field"))
 		}
 
+		if len(crt.CommonName) != 0 {
```

Comment:
I moved these checks to the top, this makes it easier to understand that these checks limit the structure of the Certificate. The following checks actually start checking the contents of the `LiteralSubject` field.